src/app/core/duplicate-records/duplicate-records.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service'; 
+import { EntityRegistry } from '../entity/database-entity.decorator';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class DuplicateRecordsService {
+  constructor(
+    private entitymapperservice: EntityMapperService,
+    private entityTypes: EntityRegistry,
+  ) {}
+  async getDataforDuplicate(data: any, schema: string) {
+   const transformedData = this.transformData(data, schema)
+   this.entitymapperservice.saveAll(transformedData);
+  }
+
+  transformData(originalData: any, schema: string): any {
+    let data = []
+    const final =[]
+    const entityConstructor = this.entityTypes.get(schema);
+    const keys =[...entityConstructor.schema.keys()]
+    const filteredKeys = keys.filter(key => key !== '_id' && key !== '_rev');
+    originalData.map((item: { record: { [x: string]: any; }; })=> {
+      delete item.record["_rev"];
+      delete item.record["_id"];
+      const strData = JSON.stringify(item.record);
+      const jsonData = JSON.parse(strData)
```

Comment:
As briefly discussed during standup:
This might be the issue why the age is not showing (I think JSON.parse just keeps a date string as a string and can't create a Date object - let alone our patched version `DateWithAge`

`Entity` class has a "copy" method (although I am not sure it will handle every potential complex property correctly):
https://github.com/Aam-Digital/ndb-core/blob/e7147b01a2cf900bfe885c5ed9bb17e11f8387c1/src/app/core/entity/model/entity.ts#L342-L352

Another possible approach (I am not sure whether this is much more than necessary, however) could be to transform the entity into "database format" and back into "entity format". This transformation using the `EntitySchemaService` makes sure that all properties are initialized as the correct object instances etc., if necessary.
For example, we use that here to store SiteSettings in localStorage and then load and transform it back:
https://github.com/Aam-Digital/ndb-core/blob/e7147b01a2cf900bfe885c5ed9bb17e11f8387c1/src/app/core/site-settings/site-settings.service.ts#L101-L102